source/helpers/attach-element.ts

```diff
@@ -0,0 +1,45 @@
+import select from 'select-dom';
+
+import hashString from './hash-string';
+
+type Position = 'append' | 'prepend' | 'before' | 'after';
+
+interface Attachment<NewElement extends Element> {
+	// eslint-disable-next-line @typescript-eslint/ban-types --  Allows dom traversing without requiring `!`
+	anchor: Element | string | undefined | null;
+	getNewElement: () => NewElement;
+	className?: string;
+	position: Position;
+	allowMissingAnchor?: boolean;
+}
+
+// Get a unique position in the code
+function getSnapshotUUID(index = 3): string {
```

Comment:
I am not 100% sure what you are doing here. But can we use https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID ?

Reply:
It's not random, it's an automatic identifier/key of the call. It's the same every time it's called (even across sessions), but it's different for every caller.

The index defines which call point is considered (from the stack).

Maybe it should just be named `callerID` 🙂 


This lets us avoid custom classes unless necessary (in this case they're necessary though)

Reply:
This was a really good lesson something I never knew.
Thanks